api/v1alpha1/replicationsource_types.go

```diff
@@ -208,6 +217,10 @@ type ReplicationSourceResticStatus struct {
 	// lastPruned in the object holding the time of last pruned
 	//+optional
 	LastPruned *metav1.Time `json:"lastPruned,omitempty"`
+	// lastUnlocked is set to the last spec.restic.unlock when a sync is done that unlocks the
+	// restic repository.
+	//+optional
+	LastUnlocked string `json:"lastUnlocked,omitempty"`
```

Comment:
Shouldn't this also be of type `metav1.Time` instead of `string`?

Reply:
No, this field works more like the spec.trigger.manual string -User supplies a unique string `spec.restic.unlock` and then after a successful unlock the status.restic.lastUnlock is updated with the same string value.  The value can then be used to decide whether unlock needs to be run on the next sync (it will only run if the previous sync has a lastUnlocked that doesn't match).  See here: https://github.com/backube/volsync/issues/653#issuecomment-1508442673